v2/client.go

```diff
@@ -102,6 +113,13 @@ func NewHTTPClient(conf HTTPConfig) (Client, error) {
 		return nil, errors.New(m)
 	}
 
+	switch conf.WriteEncoding {
+	case DefaultEncoding:
```

Comment:
These can be combined into the same switch statement.

```go
switch conf.WriteEncoding {
case DefaultEncoding, GzipEncoding:
default:
}
```

Reply:
done